### ISSUE TITLE
Compilation problem concerning simulation using MSVC (2022) on Windows 10

### ISSUE THREAD

**Issue by lenardxu** [yunshengtian/Assemble-Them-All#20]
This is from my labmate who tried to build on windows and made it work. Let me know if this helps.

"When running `python setup.py install`, you might encounter some compile errors with M_PI being undefined or something like that. To fix it, add #define _USE_MATH_DEFINES to the top of simulation/redmax/CollisionDetection/ode/odeBoxBox.cpp. In addition, comment out the line set(CMAKE_CXX_STANDARD_LIBRARIES -ldl)  in both simulation/CMakeLists.txt and simulation/redmax/CMakeLists.txt , since this library doesn't exist on Windows."

And use this environment yml instead: https://drive.google.com/file/d/1FZm8f-d3ba09MEEIJ1fQkRw7fnKZStaw/view?usp=sharing

_Originally posted by @yunshengtian in https://github.com/yunshengtian/Assemble-Them-All/issues/18#issuecomment-2054083070_

Thanks a lot for sharing the installation advice on Windows in the first place. However, when I exactly followed your procedure, there was a compilation problem, i.e., the hang on the 'Generating Code...' during the compilation. The corresponding traceback is extracted below:
<img width="959" alt="a_problem_autoplan_install_win" src="https://github.com/yunshengtian/Assemble-Them-All/assets/67544149/5b3a4cf3-92a4-4496-8f03-49737be87dbc">

To give you more details, I used Visual Studio 2022 for the compilation. And I used the release builds. 

After I googled this issue, I found two solutions: 1. disable the optimization of the compilation under the release builds; 2. switch to the debug builds. However, both options have a significant impact on the performance of simulation in the run-time. That said, the simulation runs super slow.

Hence, could you help me out there according to your experience?

**Comment by yunshengtian** [comment]
Hi thanks for reporting this issue but unfortunately I have not built simulation in Windows and I don't have much knowledge on that. The output is all warnings and no errors so I also have no clue what's going on. We suggest using mac or linux instead.

**Comment by lenardxu** [comment]
> Hi thanks for reporting this issue but unfortunately I have not built simulation in Windows and I don't have much knowledge on that. The output is all warnings and no errors so I also have no clue what's going on. We suggest using mac or linux instead.

True, there is no warning there but a long period of waiting characterized by "Generating code..." when following the specified building process. Since I moved on to other stuff, this issue is no longer that urgent. I may have to come back to this issue at some point in the future. By then look into this issue.

**Comment by gzchueng** [comment]
"So, did you manage to solve this issue later on? I’ve also been stuck at the 'Generating code...' stage."

**Comment by gzchueng** [comment]
<img width="960" height="480" alt="Image" src="https://github.com/user-attachments/assets/6d4a6c1e-8178-48d4-a959-be4e3e1464ae" />Bro, I did it on Windows. Originally, python setup.py install compiled by default as release, and the compilation process would show the ‘Generating code...’ issue. Later, I used gpt to modify the internal cmakelist file and setup.py file, changed the compilation process to debug compilation, and used python setup.py build --debug install, and it worked. If you need the modified file, please contact me at gzchueng@gmail.com.
